e2e-tests/spring-rest-openapi-v2/src/test/java/org/evomaster/e2etests/spring/examples/displayissue/DisplayIssueEMTest.java

```diff
@@ -0,0 +1,144 @@
+
+
+                    Solution<RestIndividual> solutionWhite = initAndRun(args);
+
+                    // number of covered targets
+                    // the output should contain
+                    // * Covered targets (lines, branches, faults, etc.): {same value as the number of covered targets
+                    // from statistics} -
+                    // * Potential faults: {same value as the number of potential faults from statistics}
+                    String expectedNumberOfPotentialFaultsWhite = findValueOfItemWithKeyInStats(solutionWhite, "potentialFaults");
+                    String expectedNumberOfCoveredTargetsWhite = findValueOfItemWithKeyInStats(solutionWhite, "coveredTargets");
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Covered targets " +
+                            "(lines, branches, faults, etc.): " + expectedNumberOfCoveredTargetsWhite));
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Potential faults: "
+                            + expectedNumberOfPotentialFaultsWhite));
+
+                    // now run the same thins as a blackbox test and ensure that
+                    byteArrayOutput = new ByteArrayOutputStream();
+                    System.setOut(new PrintStream(byteArrayOutput));
+
+                    args.clear();
+
+                    args.add("--blackBox");
+                    args.add("true");
+                    args.add("--bbTargetUrl");
+                    args.add(baseUrlOfSut);
+                    args.add("--bbSwaggerUrl");
+                    args.add(baseUrlOfSut + "/v2/api-docs");
+                    args.add("--outputFormat");
+                    args.add("JAVA_JUNIT_4");
+                    args.add("--writeStatistics");
+                    args.add("true");
+
+                    Solution<RestIndividual> solutionBlack = initAndRun(args);
+
+                    // number of covered targets
+                    // the output should contain
+                    // * Covered targets (lines, branches, faults, etc.): {same value as the number of covered targets
+                    // from statistics} -
+                    // * Potential faults: {same value as the number of potential faults from statistics}
+                    String expectedNumberOfPotentialFaultsBlack = findValueOfItemWithKeyInStats(solutionBlack, "potentialFaults");
+                    String expectedNumberOfCoveredTargetsBlack = findValueOfItemWithKeyInStats(solutionBlack, "coveredTargets");
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Covered targets " +
+                            "(lines, branches, faults, etc.): " + expectedNumberOfCoveredTargetsBlack));
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Potential faults: "
+                            + expectedNumberOfPotentialFaultsBlack));
+
+                }
+        );
+
+        System.setOut(old);
```

Comment:
if test fail, then `setOut` will never be called. Rather move this logic in a `AfterEach` method

Reply:
Thank you so much for your comment Andrea, I have put the two comments inside finally to make sure it always gets called. I have also added another line removing the `statistics.csv` file.